server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -211,12 +221,17 @@ private void confirmDeletes(GarbageCollectionEnvironment gce,
 
         dir = makeRelative(dir, 2);
 
-        if (candidateMap.remove(dir) != null)
+        if (candidateMap.remove(dir) != null) {
           log.debug("Candidate was still in use: " + dir);
+        }
+      } else if (TabletsSection.TabletColumnFamily.PREV_ROW_COLUMN.hasColumns(key)) {
+        readTracker.sawPrevRow();
       } else
         throw new RuntimeException(
             "Scanner over metadata table returned unexpected column : " + entry.getKey());
     }
+    // process last row to check metadata read included the dir entry.
+    readTracker.sawLastRow();
```

Comment:
The name of this method is a bit confusing. It implies positive affirmation that the last row possible was actually seen. But that's what it's supposed to be verifying. So, maybe rename it like:

```suggestion
    readTracker.checkLastRow();
```

Or just call it `close` and make it AutoCloseable and use it with try-with-resources.

Reply:
Did some general renaming in bb4f078d46.